README.md

```diff
@@ -25,6 +25,71 @@ Laboratory](https://dai.lids.mit.edu/) at MIT.
 The benchmarking of a synthesizer is a process in which different datasets are generated by your
 synthesizer. Then, each couple of real and synthetic data is evaluated with multiple scores.
 
+## What is a synthesizer function?
+
+In order to use **SDGym** you will need a synthesizer function.
+This is a function that takes as input a numpy matrix with real data and outputs another numpy
+matrix with the same shape filled with synthesized data.
+
+Also, alongside the real data, some additional variables informing about the column contents
+will be passed, which means that the exact signature of the function will be like this:
+
+
+```python
+def my_synthesizer_function(
+    real_data: numpy.ndarray,
+    categorical_columns: list,
+    ordinal_columns: list
+) -> syntehtesized_data: numpy.ndarray
+```
+
+If your synthesizer implements a different interface, you can wrap it in a function like this:
+
+```python
+def my_synthesizer_function(real_data):
+    # ...do all necessary steps here...
+    return synthesized_data
+```
+
+This function should contain inside it all the parameters and arguments needed to use your
+synthesizer and call it to generate the new synthesized data based on the real data that is
+being passed.
+
+## What data should your synthesizer work with?
+
+As we mentioned in the section before, the main input of **SDGym** is a synthesizer to be
+benchmarked, which is expected to be a function that has as unique input and output a table of
+data.
+
+The inputs for your synthesizer funciton should be:
+
+* `real_data`: a 2D `numpy.ndarray` with the real data the your synthesizer will attempt to imitate.
+* `categorical_columns`: a `list` with the indexes of any columns that should be considered
+  categorical independently on their type.
+* `ordinal_columns`: a `list` with the indexes of any integer columns that should be treated as
+  ordinal values.
+
+And the output should be a single 2D `numpy.ndarray` with the exact same shape as the `real_data`
+matrix.
+
+## Demo Synthesizers
+
+In order to get started using the benchmarking tool, some demo Synthesizers have been included
+in the library.
+
+These synthesizers can be imported from the `sdgym.synthesizers` module and are classes that have
+the following methods:
+
+* `fit`: Fits the synthesizer on the data. Expects the following arguments:
+    * `data (numpy.ndarray)`: 2 dimensional Numpy matrix with the real data to learn from.
+    * `categorical_columns (list or tuple)`: List of indexes of the columns that are categorical within the dataset.
+    * `ordinal_columns (list or tuple)`: List of indexes of the columns that are ordinal within the dataset.
+* `sample`: Generates new data resembling the original dataset. Expects the following arguments:
+    * `n_samples (int)`: Number of samples to generate.
+* `fit_sample`: Fits the synthesizer on the dataset and then samples as many rows as there were in
+  the original dataset.
+
```

Comment:
Maybe move line 49 here. Give a solid example on how to wrap existing synthesizer class to a synthesizer function. I would prefer using Synthesizer class instead of synthesizer function for two reasons: 
- Usually, a synthesizer requires 2 phases, fit and sample.
- It's possible to make a consistent save/restore API.

Reply:
> Maybe move line 49 here.

No, this function needs to stay there, right where the synthesizer function is explained, as that  is the place where the clarification about how to use a different API needs to be. Otherwise, that explanation would be really confusing.

> Give a solid example on how to wrap existing synthesizer class to a synthesizer function.

The complete example about how to create an instance of a synthesizer and evaluate it using the benchmark function is further down, on the usage section, and I'd also prefer to keep it there, as adding some code examples before the actual basic benchmark usage example would also be very confusing.

However, what I did is change a bit the Usage section to also show, apart from the benchmark, how the `fit` and `sample` methods would be used on their own without running the benchmarking.

> I would prefer using Synthesizer class instead of synthesizer function for two reasons:
> 
> * Usually, a synthesizer requires 2 phases, fit and sample.
> * It's possible to make a consistent save/restore API.

This is already like that: the synthesizers included are all classes and they all have the `fit` and `sample` methods. They just provide the `fit_sample` method for convenience, to have something with the necessary signature to be able to be passed directly as an argument to the `benchmark` function without having to do any wrapping.

But this is slightly different than the `benchmark` function. This has been created as simple as possible to support as many different APIs as possible, and that's why it has been decided to make it expect a function as input: it's much easier to wrap a Class based synthesizer inside a function than to make any synthesizer that is not a class work as a class and support the fit + sample interface.